src/main/java/jewellery/inventory/mapper/UserMapper.java

```diff
@@ -6,11 +6,12 @@
 import jewellery.inventory.model.User;
 import org.mapstruct.Mapper;
 
-@Mapper(
-    componentModel = "spring")
+@Mapper(componentModel = "spring")
 public interface UserMapper {
   UserRequestDto toUserRequest(User user);
 
+  User toUserEntity(UserResponseDto userResponseDto);
```

Comment:
We shouldn't need this method. We can create a getUser method in UserService that returns the entity. In the service the current method that fetches the user is fetching an entity and mapping it to a ResponseDto. We can do the same as we have done for fetching the product